raft-tech/TANF-app
# Issue 639: As an OFA Admin, I need to be able to access the admin functions in TDP

### Summary of Changes

This PR makes it so that when a user is NOT a `System Admin`, they will not see the `Admin` tab in the TANF app navigation.

Addresses issue #639


### How to Test

- Sign in to the frontend with any user.
- Ensure the Admin tab is not visible for your (non-System Admin) user.
	- If the Admin tab _is_ visible, your user may already have the `System Admin` admin. Sign into Django Admin at `localhost:8080/admin` and [check the roles for your user](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/Technical-Documentation/user_role_management.md#user-role-management).
	- If the Admin tab _is not_ visible, follow the steps linked above to get to the User Edit menu, and add the `System Admin` role to your user.
		- If you do not see the `System Admin` role in the `Groups` list, create a new group with the `System Admin` name.
		- Make sure to select the the role in the list (it should be highlighted), and then click "Save".
		- If you are not able to sign into Django Admin, you need to add both the `superuser` and `staff` roles to your user [through the Django shell](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/Technical-Documentation/user_role_management.md#-updating-user-roles-through-the-cli).
- Ensure clicking on the Admin tab takes you to Django Admin.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] System Admin users can see and click the Admin tab
  - [x] Non-System Admin users do not see an Admin tab within the application
  - [x] Admin tab links to Django Admin (`<BACKEND-URL>/admin`) instead of a blank page

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Backend coverage: 100% (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [x] Did automated and manual testing with @jorgegonzalez and @reitermb  using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Do manual code review and manual testing detect any security issues? *No issues detected*
